packages/eas-cli/src/build/build.ts

```diff
@@ -274,7 +282,33 @@ async function uploadProjectAsync<TPlatform extends Platform>(
             completedMessage: (duration: string) => `Uploaded to EAS ${chalk.dim(duration)}`,
           })
         );
-        return bucketKey;
+        let projectMetadataFile: any;
+        try {
+          projectMetadataFile = await makeProjectMetadataFileAsync(projectTarball.path);
+
+          const metadataLocation = await uploadFileAtPathToGCSAsync(
+            ctx.graphqlClient,
+            UploadSessionType.EasBuildGcsProjectSources,
+            projectMetadataFile.path,
+            createProgressTracker({
+              total: projectMetadataFile.size,
+              message: ratio =>
+                `Uploading metadata to EAS Build (${formatBytes(
+                  projectMetadataFile.size * ratio
+                )} / ${formatBytes(projectMetadataFile.size)})`,
+              completedMessage: (duration: string) => `Uploaded to EAS ${chalk.dim(duration)}`,
+            })
+          );
+          return { bucketKey, metadataLocation };
+        } catch (e) {
+          Log.warn('Failed to upload metadata to EAS Build');
+          Log.debug(e);
+          return { bucketKey };
+        } finally {
+          if (projectMetadataFile) {
+            await fs.remove(projectMetadataFile.path);
+          }
+        }
```

Comment:
```suggestion
```

Reply:
...and below top-level `try-catch` block:
```ts
        let projectMetadataFile: any;
        try {
          projectMetadataFile = await makeProjectMetadataFileAsync(projectTarball.path);
          const metadataLocation = await uploadFileAtPathToGCSAsync(
            ctx.graphqlClient,
            UploadSessionType.EasBuildGcsProjectSources,
            projectMetadataFile.path,
            createProgressTracker({
              total: projectMetadataFile.size,
              message: ratio =>
                `Uploading metadata to EAS Build (${formatBytes(
                  projectMetadataFile.size * ratio
                )} / ${formatBytes(projectMetadataFile.size)})`,
              completedMessage: (duration: string) => `Uploaded to EAS ${chalk.dim(duration)}`,
            })
          );
          return { bucketKey, metadataLocation };
        } catch (e) {
          Log.warn('Failed to upload metadata to EAS Build');
          Log.debug(e);
          return { bucketKey };
        } finally {
          if (projectMetadataFile) {
            await fs.remove(projectMetadataFile.path);
          }
        }
 ```
 so the `try-catch` aren't nested making it easier to read